ncm-cron/src/main/perl/cron.pm

```diff
@@ -453,76 +511,85 @@ sub cleanCrontabs {
+                    group => "sys");
+            my $outlines = "";
+            my $foundNCM = 0;
+            foreach my $line (split("\n", "$fh")) {
+                $line =~ /$NCM_START/ && do {$foundNCM = 1};
+                $outlines .= "$line\n" if !$foundNCM;  # Don't save ncm lines
+                $line =~ /$NCM_STOP/ && do {$foundNCM = 0};
+                # If an old style ncm managed cron file, simply delete all content.
+                if ($self->isOldSolarisFile($path, $line)) {
+                    $outlines = "";
+                    last;
                 }
-                $self->debug(2, "Keeping lines:\n$outlines\n") if $outlines;
-                $fh->set_contents($outlines);  # Save non NCM-CRON lines
-                $fh->seek(0, 2);  # eof
-                print $fh "$ncmStart $ncmMsg\n"  # Set NCM-CRON START:
-                    if $configType eq "Configure";
-                $self->warn("NCM-CRON END: missing from $fPath. Deleting to EOF")
-                    if $foundNCM eq 1;
             }
-        }
-        closedir DIR;
-    # Linux: collect the current entries managed by ncm-cron in the cron.d
-    # directory and delete those no longer part of the configuration.
-    # Returns an empty hash (return value unused in Linux context).
-    } else {
-        opendir DIR, $linux_crondir;
-        while (my $to_unlink = readdir DIR) {
-            next if $to_unlink !~ /$cron_entry_regexp$/;
-            next if defined($cronEntryNames{$to_unlink});
-            $to_unlink = "$linux_crondir/$to_unlink";
+            $self->debug(2, "Keeping lines:\n$outlines\n") if $outlines;
+            $fh->set_contents($outlines);  # Save non NCM-CRON lines
+            $fh->seek(0, 2);  # eof
+            print $fh "$NCM_START $NCM_MSG\n"  # Set NCM-CRON START:
+                if $configType eq "Configure";
+            $self->warn("NCM-CRON END: missing from $path. Deleting to EOF")
+                if $foundNCM eq 1;
+        } else {
+            # Linux: collect the current entries managed by ncm-cron in the cron.d
+            # directory and delete those no longer part of the configuration.
+            # Returns an empty hash (return value unused in Linux context).
+            my $cron_entry_regexp = $CRON_ENTRY_EXTENSION;
+            $cron_entry_regexp =~ s/\./\\./g;
+
+            next if $filename !~ /$cron_entry_regexp$/;
+            next if defined($cronEntryNames{$filename});
+
             # untainted to_unlink to work with tainted perl mode (-t option)
-            if ($to_unlink =~ m{^($linux_crondir/.*$cron_entry_regexp)$}) {
-                $to_unlink = $1;  # $to_unlink is now untainted
+            if ($path =~ m{^($CRONDIR_LINUX/.*$cron_entry_regexp)$}) {
+                my $to_unlink = $1;  # $to_unlink is now untainted
+                # This used to be done with LC::Check::absence(file => True)
+                # but this is a loop over files only, so it's safe to use CAF::Path::cleanup
```

Comment:
Are the consequences of a directory ending up in here so bad that it should be explicitly checked whether it's a directory or a file before passing to cleanup?